application/src/main/java/org/togetherjava/tjbot/features/moderation/scam/ScamBlocker.java

```diff
@@ -141,6 +137,10 @@ public void onMessageReceived(MessageReceivedEvent event) {
             isSafe = false;
         }
 
+        if (isSafe && similarMessagesDetector.doSimilarMessageCheck(event)) {
```

Comment:
`doSimilarMessageCheck` should be renamed. maybe `hasPostedSimilarMessageTooOften(event)`

if you dont want it to start with a question-word bc of the side effects, maybe `handleHasPostedSimilarMessagesTooOften(event)`.

but the name should clarify what the `boolean` it returns means and how it fits into the logic of `ScamBlocker`